Cairo Desktop/Cairo Desktop/TaskButton.xaml.cs

```diff
@@ -167,28 +170,32 @@ private void setToolTip()
             }
         }
 
-        private void setIconSize()
+        private void setLabelVisibility()
```

Comment:
In this method, rather than manipulating the grid, you can just change the visibility of the label. For example: `WinTitle.Visibility = Settings.Instance.HideTaskbarLabels ? Visibility.Collapsed : Visibility.Visible;`

Reply:
Yes, changing the visibility is enough and may be a cheaper operation layout wise.